DS3: Add DLC, DLC Progressive Locations, and add Weapon Levels Options

Br00ty opened the pull request:
Please format your title with what portion of the project this pull request is
targeting and what it's changing.

ex. "MyGame4: implement new game" or "Docs: add new guide for customizing MyGame3"

## What is this fixing or adding?
Adds DLC, DLC Progressive locations, and add adjustable Weapon Level Options

## How was this tested?
Through a AP test and through friends. 

## If this makes graphical changes, please attach screenshots.

eldsmith committed:
DS3: Add more options for weapon upgrade randomization

Add range for possible upgrades, as well as setting percentage of randomized weapons

eldsmith committed:
Merge branch 'main' into ds3-weapon-randomization

eldsmith committed:
Merge branch 'main' into ds3-weapon-randomization

Berserker66 committed:
Tests: make sure DB overwrite actually takes

el-u committed:
lufia2ac: document changes from the vanilla game

Berserker66 committed:
WebHost: fix LttP tracker crash

toasterparty committed:
[OC2] Horde Levels Logically Give 1-Star (#1322)

toasterparty committed:
[OC2] Remove DLC items from item pool if vanilla level order (#1323)

toasterparty committed:
[OC2] New Option - Shorter Horde Levels (#1328)

Br00ty committed:
update DS3 description

changed "items" to "item"

Br00ty committed:
Merge branch 'ArchipelagoMW:main' into ds3-quick-fixes

Br00ty committed:
Update __init__.py

change 'required_client_version' back to 0.3.6 to work with DLL

PoryGone committed:
SMW: Adjust Butter Bridge 2 - Dragon Coins logic (#1332)

el-u committed:
lufia2ac: prevent using party member items if party is full

Berserker66 committed:
Subnautica: 2.0 compatibility (#1329)

el-u committed:
lufia2ac: rename starting_capsule/starting_party options to default_capsule/default_party

Alchav committed:
[Pokemon R/B] Allow 0 exp setting and add logic rule to Cinnabar Gym to ensure higher level Pokémon are catchable

Alchav committed:
[Pokemon R/B] add logic to Fighting Dojo and check for non-vanilla old_man setting for cinnabar gym

Br00ty committed:
Merge pull request #17 from FridbergT/ds3-weapon-randomization

DS3: Additional Weapon Randomization Settings from FridbergT

Br00ty committed:
Merge branch 'ArchipelagoMW:main' into ds3-quick-fixes

Berserker66 committed:
Core: Make apworlds function mostly before Python 3.10

espeon65536 committed:
Ocarina of Time 0.3.7 hotfixes (#1336)

- Since multiprocessing and exceptions apparently don't interact well, any exceptions in `generate_output` are handled more gracefully now. The event is always set as well, so the process will no longer hang on an exception there.
- The triforce object file was renamed in 7.0. I didn't change the filename in the code, so it would crash on non-Windows systems.
- Some hint distributions just crash, so I'm temporarily removing them. It will take a while to port the relevant functionality and I'd rather not hold up the next version release.

espeon65536 committed:
oot: repair closed forest + dungeon ER

Berserker66 committed:
Subnautica: fix early seaglide

Alchav committed:
[Pokemon R/B] actually implement lose_money_on_blackout

Alchav committed:
[Pokemon R/B] Fix missing lift key logic

Alchav committed:
[Pokémon R/B] Fix DeathLink softlock and increment data version (#1348)

Br00ty committed:
Merge branch 'ArchipelagoMW:main' into ds3-quick-fixes

Berserker66 committed:
Network: allow sending frozenset

Alchav committed:
Pokemon R/B: Add Revive to better_shops (#1352)

Jarno458 committed:
WebHost: Fixed game order by title in Site Map (#1349)

Commit:
WebHost: Capitalize Special Range choices to keep consistency. (#1360)

Commit:
Rogue Legacy: Fix early vendors and architect... again. (#1359)

alwaysintreble committed:
MultiServer/WebHost: rename all references to forfeit and deprecate it (#1243)

* Webhost: rename all references to forfeit and deprecate it

* needed some renames in multiserver for all the commands to function

* remove forfeit commands

* support forfeit_mode for clients

* rename `forfeit_player` to `release_player`

alwaysintreble committed:
Generate: fix default utils options (#1361)

Berserker66 committed:
Clients: retry connection with ssl (#1341)

Alchav committed:
Pokémon R/B: Fix Pokémon Tower 7F crash (#1362)

Commit:
Docs: Add special name keywords to docs. (#1353)

Commit:
MultiServer: Remove `forced_auto_forfeit` (#1363)

toasterparty committed:
[OC2] Fix "Moon 1-5" never appearing in level pool (#1366)

JoshuaEagles committed:
Subnautica: add Linux note to setup guide (#1365)

beauxq committed:
Zillion: minor terrain logic update

standing on a moving walkway requires 2 columns of standing space

toasterparty committed:
[OC2] Logic: fixes fails when horde levels/items are excluded from location pool (#1369)

JoshuaEagles committed:
SA2B: Add Linux section to setup guide (#1374)

t3hf1gm3nt committed:
update LTTP player template to add all universal AP options (#1372)

Jarno458 committed:
Sudoku: Hints will no-longer duplicate (#1371)

Berserker66 committed:
LttP: attempt at preventing ghost location checks (#1355)

eudaimonistic committed:
Docs: code_of_conduct.md (#1350)

Update to point of contact.

beauxq committed:
ZillionClient: fix manual disconnect (#1266)

espeon65536 committed:
Ocarina of Time: 0.3.7 hotfixes round 2 (#1351)

* oot: repair closed forest + dungeon ER

* oot: finally skip triforce pieces in balancing

* oot: fix mq_dungeons_mode set to mq or count

* oot: force 0.3.7 client
hopefully this makes people update

* oot: temp fix for skip-child-zelda crash
eventually I want to decide on a better fix for this though

* oot: remove skip-child-zelda item inside if tree

* oot: fix classification of some thieves hideout locations in tracker

* oot: fix regional shuffle for hideout keys and ganon boss key

* oot: properly attach hints to dungeon locations

* Fix entrance shuffle flag not being set correctly due to new dungeon shuffle option format

Br00ty committed:
last fixes for DLC

fixed up last minute DLC details, added option for Weapon Levels

Br00ty committed:
Merge branch 'ArchipelagoMW:main' into ds3-next-feature

espeon65536 committed:
oot: force itempool to higher settings if required by heart logic

Alchav committed:
Pokémon R/B: Fix Safari Zone Gate bug (#1381)

el-u committed:
alttp: add item rules for prize locations (#1380)

Marechal-L commented on worlds/dark_souls_3/__init__.py:
```diff
@@ -175,7 +178,7 @@ def create_regions(self):
             self.multiworld.get_entrance("Goto Ringed City", self.player).connect(ringed_city_region)
 
-    # For each region, add the associated locations retrieved from the corresponding location_table
+    # For each region, add the associated locations retrieved from the corresponding =
```
Is it a mistake or you wanted to write something else ?

Br00ty replied:
this was a typo, changed it back

Br00ty committed:
Merge branch 'ArchipelagoMW:main' into ds3-next-feature

Marechal-L commented on worlds/dark_souls_3/Options.py:
```diff
@@ -29,10 +29,56 @@ class NoEquipLoadOption(Toggle):
+    option_epic = 3
+
+
+class RandomizeWeaponsLevelPercentageOption(Range):
```
This option is missing a display_name.
We should add display_name = "Percentage of randomized weapons" for example

Br00ty replied:
you are right! added the display name that we needed

Br00ty committed:
quick fixes

fixed a typo, and added a display_name to RandomizeWeaponsLevelPercentageOption in Options.py
